src/toir.c

```diff
@@ -148,7 +148,33 @@ elem *getEthis(Loc loc, IRState *irs, Dsymbol *fd)
              * adding this frame into the linked list of stack
              * frames.
              */
-            if (thisfd->hasNestedFrameRefs())
+            if (fdp != thisfd)
+            {
+                //printf("L%d fd = %s, fdparent = %s, fd->toParent2() = %s\n",
+                //    __LINE__, fd->toPrettyChars(), fdparent->toPrettyChars(), fdp->toPrettyChars());
+                assert(fd->ident == Id::require || fd->ident == Id::ensure);
+                assert(thisfd->hasNestedFrameRefs());
```

Comment:
This branch needs a small comment like the `thisfd->hasNestedFrameRefs()` branch below.